src/app/shared/components/validation-hint/validation-hint.component.ts

```diff
@@ -74,24 +83,24 @@ export class ValidationHintComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   private checkValidationErrors(errors: ValidationErrors): void {
-    this.invalidEmail = !!errors?.email;
+    this.invalidEmail = errors?.email;
     if (this.isPhoneNumber) {
-      this.invalidPhoneLength = !!errors?.minlength && !errors?.maxlength;
+      this.invalidPhoneLength = errors?.minlength && !errors?.maxlength;
     } else if (this.isEdrpouIpn) {
-      this.invalidEdrpouIpn = !!errors?.minlength && !errors?.maxlength;
+      this.invalidEdrpouIpn = errors?.minlength && !errors?.maxlength;
     } else {
-      this.invalidFieldLength = !!(errors?.maxlength || errors?.minlength);
+      this.invalidFieldLength = errors?.maxlength || errors?.minlength;
     }
   }
 
   private checkInvalidText(errors: ValidationErrors): void {
-    const requiredPattern = errors?.pattern?.requiredPattern;
+    const requiredPattern = errors?.pattern?.requiredPattern?.toString();
 
-    this.invalidSymbols = NAME_REGEX === requiredPattern;
-    this.invalidCharacters = NO_LATIN_REGEX === requiredPattern;
-    this.invalidStreet = STREET_REGEX === requiredPattern;
-    this.invalidHouse = HOUSE_REGEX === requiredPattern;
-    this.invalidSectionName = SECTION_NAME_REGEX === requiredPattern;
+    this.invalidSymbols = NAME_REGEX.toString() === requiredPattern || FULL_NAME_REGEX.toString() === requiredPattern;
+    this.invalidCharacters = NO_LATIN_REGEX.toString() === requiredPattern;
+    this.invalidStreet = STREET_REGEX.toString() === requiredPattern;
+    this.invalidHouse = HOUSE_REGEX.toString() === requiredPattern;
+    this.invalidSectionName = SECTION_NAME_REGEX.toString() === requiredPattern;
   }
```

Comment:
There should be some confusion about using `toString()` and it's because constants regexes are `RegExp` objects, while `requiredPattern` usually `string` (excluding tests).